contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {AccessControlUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import {PercentMath} from "../../lib/PercentMath.sol";
+import {IStrategy} from "../IStrategy.sol";
+import {CustomErrors} from "../../interfaces/CustomErrors.sol";
+import {IVault} from "../../vault/IVault.sol";
+import {IStabilityPool} from "../../interfaces/liquity/IStabilityPool.sol";
+import {ERC165Query} from "../../lib/ERC165Query.sol";
+
+/***
+ * Liquity Strategy generates yield by investing LUSD assets into Liquity Stability Pool contract.
+ * Stability pool gives out LQTY & ETH as rewards for liquidity providers.
+ * The LQTY rewards are normal yield rewards
+ * But the Stability Pool achievs ETH rewards by Liquidating Troves using the LUSD we deposited.
+ * So our balance of LUSD goes down and we get an 1.1x (or higher) value of ETH. In short, we make a 10% profit in ETH everytime our LUSD is used for liquidation by the stability pool
+ * the harvest method here withdraws those LQTY & ETH rewards, swaps them into LUSD and then deposits them back to the stability pool.
+ * we should make sure to harvest at regular intervals because if the value of ETH rewards goes below 1x of the LUSD used for liquidation then we will make a net loss on our LUSD.
+ * the contract uses 0xapi for swapping the tokens.
+ */
+contract LiquityStrategy is
+    IStrategy,
+    UUPSUpgradeable,
+    AccessControlUpgradeable,
+    CustomErrors
+{
+    using PercentMath for uint256;
+    using ERC165Query for address;
+
+    error StrategyStabilityPoolCannotBeAddressZero();
+    error StrategyYieldTokenCannotBe0Address();
+    error TokenApprovalFailed(address token);
+    error TokenTransferFailed(address token);
```

Comment:
Why aren't these `StategySomething`?

Reply:
yes that can be changed. Does StrategyTokenApprovalFailed & StrategyTokenTransferFailed work?